ionoscloud/resource_server.go

```diff
@@ -812,6 +869,107 @@ func resourceServerUpdate(ctx context.Context, d *schema.ResourceData, meta inte
 	dcId := d.Get("datacenter_id").(string)
 	request := ionoscloud.ServerProperties{}
 
+	server, _, err := client.ServersApi.DatacentersServersFindById(ctx, dcId, d.Id()).Execute()
+	currentVmState := server.Properties.VmState
+	serverType := server.Properties.Type
+	if err != nil {
+		diags := diag.FromErr(fmt.Errorf("could not retrieve server vmState: %s", err))
+		return diags
+	}
+	if *currentVmState == "SUSPENDED" && !d.HasChange("vm_state") {
+		diags := diag.FromErr(fmt.Errorf("cannot update a suspended Cube Server, must change the state to RUNNING first"))
+		return diags
+	}
+
+	if d.HasChange("vm_state") {
+		_, newState := d.GetChange("vm_state")
+		newVmState := newState.(string)
+		switch *serverType {
+		case "ENTERPRISE":
```

Comment:
Let's move this to something like stopServer(serverType) and startServer(serverType) and have the logic in there.

Reply:
refactored into the server service